### PR TITLE
Feature/share based on origin in es

### PR DESCRIPTION
**Pull request by kwarunek**

### CODE REVIEW

File: src/main/java/net/iponweb/disthene/reader/service/index/IndexService.java [hunk 1]
```diff
@@ -41,44 +44,42 @@ public IndexService(IndexConfiguration indexConfiguration) {
         }
     }
 
-    public List<String> getPaths(String tenant, List<String> wildcards) {
+    public Map<String, String> getPaths(String tenant, List<String> wildcards) {
```
Comment (kwarunek) [inline src/main/java/net/iponweb/disthene/reader/service/index/IndexService.java]: To be fully compliant with tenant feature (and use it in the future) these function should return some `Path` object with respective `getTenant` and `getPath` . But for now I'll leave it as is.

File: src/main/java/net/iponweb/disthene/reader/service/index/IndexService.java [hunk 2]
```diff
@@ -41,44 +44,42 @@ public IndexService(IndexConfiguration indexConfiguration) {
...
-            if (WildcardUtil.isPlainPath(wildcard)) {
-                result.add(wildcard);
-            } else {
-                regExs.add(WildcardUtil.getPathsRegExFromWildcard(wildcard));
-            }
+            regExs.add(WildcardUtil.getPathsRegExFromWildcard(wildcard));
         }
+        String regEx = Joiner.on("|").skipNulls().join(regExs);
 
-        logger.debug("getPaths plain paths: " + result.size() + ", wildcard paths: " + regExs.size());
```
Comment (bkalka) [inline src/main/java/net/iponweb/disthene/reader/service/index/IndexService.java]: debugi można zostawić, ale przerobić z "+" na formatowanie które nie będzie interpretowane za każdym wywołaniem funkcji

File: src/main/java/net/iponweb/disthene/reader/service/index/IndexService.java [hunk 3]
```diff
@@ -41,44 +44,42 @@ public IndexService(IndexConfiguration indexConfiguration) {
...
+                .execute().actionGet();
 
-            SearchResponse response = client.prepareSearch(indexConfiguration.getIndex())
-                    .setScroll(new TimeValue(indexConfiguration.getTimeout()))
-                    .setSize(indexConfiguration.getScroll())
-                    .setQuery(QueryBuilders.filteredQuery(QueryBuilders.regexpQuery("path", regEx),
-                            FilterBuilders.termFilter("tenant", tenant)))
-                    .addField("path")
-                    .execute().actionGet();
+        // if total hits exceeds maximum - abort right away returning empty array
```
Comment (bkalka) [inline src/main/java/net/iponweb/disthene/reader/service/index/IndexService.java]: i co się wtedy dzieje jak pójdzie pusta zwrotka?

File: src/main/java/net/iponweb/disthene/reader/service/index/IndexService.java (hunk 2 above)
Comment (kwarunek) [inline src/main/java/net/iponweb/disthene/reader/service/index/IndexService.java]: There is an issue (https://github.com/EinsamHauer/disthene-reader/issues/13) for that. Even more I would upgrade log4j and target against java 8. But currently, these tasks are out of the scope.

File: src/main/java/net/iponweb/disthene/reader/service/index/IndexService.java (hunk 3 above)
Comment (kwarunek) [inline src/main/java/net/iponweb/disthene/reader/service/index/IndexService.java]: `return result` - result is an empty map at this point

**Pull request merged**